learningresources/api.py

```diff
@@ -120,13 +103,93 @@ def type_id_by_name(name):
     return obj.id
 
 
-def get_repos(user):
+def get_repos(user_id):
     """
     Get all repositories a user may see.
 
     Args:
         user (auth.User): request.user
     Returns:
-        repos query set of learningobject.Repository: repositories
+        repos query set of learningresource.Repository: repositories
+    """
+    return Repository.objects.filter(created_by__id=user_id).order_by('name')
+
+
+def get_repo_courses(repo_id):
+    """
+    Get courses for a repository.
+    Args:
+        repository (int): pk of learningresource.Repository
+    Returns:
+        courses (queryset of learningresource.Course): courses
+    """
+    return Course.objects.filter(repository__id=repo_id)
+
+
+def create_repo(name, description, user_id):
+    """
+    Create a new repository.
+    Args:
+        name (unicode): repository name
+        description (unicode): repository description
+        user_id (int): user ID of repository creator
+    Returns:
+        repo (learningresources.Repository): newly-created repo
+    """
+    with transaction.atomic():
+        return Repository.objects.create(
+            name=name, description=description,
+            created_by_id=user_id,
+        )
+
+
+def get_courses(repo_id):
+    """
+    Get all user's courses.
+    Args:
+        repo_id (int): primary key of the repository
+    Returns:
+        Queryset of learningresources.Course: courses
+    """
+    return Course.objects.filter(repository_id=repo_id)
+
+
+def get_runs(repo_id):
+    """
+    Get runs in all user's courses for the repo.
+    Args:
+        repo_id (int): primary key of the repository
+    Returns:
+        runs (list of strings): run names
+    """
+    courses = get_courses(repo_id)
+    return sorted(list(set([x.run for x in courses])))
```

Comment:
If I understand that properly, then no, because the same run may be in multiple courses, and we just want unique runs. I think this would be simpler with a raw SQL query, like

```sql
SELECT DISTINCT(run) FROM learningresources_course ORDER BY run
```

Maybe I should do that, if nobody objects to the use of `objects.raw()`.